modules/tinymce/src/plugins/table/main/ts/ui/CellDialog.ts

```diff
@@ -88,19 +109,37 @@ const onSubmitCellForm = (editor: Editor, cells: HTMLTableCellElement[], api) =>
   });
 };
 
+const getData = (editor: Editor, startCell: SugarElement<Element>, cells: SugarElement<HTMLTableCellElement>[]) => {
+  const table = TableLookup.table(startCell).getOrNull();
+
+  const warehouse = Warehouse.Warehouse.fromTable(table);
```

Comment:
This and the following lines are a duplicate of the above... can you please remove the duplication here?

Reply:
It honestly doesn't quite seem worth it.
`columns` are used in widely different ways, so that needs to stay.
Because of the above `warehouse` also needs to stay.
Which only leaves the `allCells` and the filtering. Except that the filter is also different, needing a different set of comperators. We'd pull out three-four lines and still have to put in a lot of stuff to the function.

Reply:
I count 12 duplicate lines, so that definitely matters. Even if it was 4 lines, we should not be creating duplicate code (DRY principle) as it leads to maintenance nightmares down the line, as if you change one you need to change both.

Reply:
This hasn't been resolved, there's still a lot of duplication here.

Reply:
What duplication? The only duplication I can see here is the stuff that's used by later, different code and thus can't be extracted.

Reply:
Looking up the table from the cell, creating the warehouse, etc...

Reply:
Yes. Getting the columns requires the warehouse, which requites the table, which requites looking it up from the cell. As I said in my first answer. Did you want it to return some kind of tuple?

Reply:
As discussed today, this can be refined to return the cell and column, as that's the data we always use. Here's a rough example:

````
const getSelectedCells = (cells: SugarElement<HTMLTableCellElement>[]) =>
  TableLookup.table(cells[0]).map((table) => {
    const warehouse = Warehouse.fromTable(table);

    const allCells = Warehouse.justCells(warehouse);

    const filtered = Arr.filter(allCells, (cellA) =>
      Arr.exists(cells, (cellB) =>
        Compare.eq(cellA.element, cellB)
      )
    );

    return Arr.map(filtered, (cell) => ({
      element: cell.element,
      column: Warehouse.getColumnAt(warehouse, cell.column)
    }));
  });
```